nsis/src/nsis/file/NsisExecutable.java

```diff
@@ -59,15 +63,12 @@ public static NsisExecutable createNsisExecutable(GenericFactory factory, BytePr
 
 	private void initNsisExecutable(GenericFactory factory, ByteProvider bp, SectionLayout layout)
 			throws IOException, InvalidFormatException {
-		this.reader = new FactoryBundledWithBinaryReader(factory, bp, /* isLittleEndian= */ true);
+		this.reader = new FactoryBundledWithBinaryReader(factory, bp,
+				NsisConstants.IS_LITTLE_ENDIAN);
 		this.headerOffset = findHeaderOffset();
 		initScriptHeader();
-		if ((this.scriptHeader.compressedHeaderSize & FLAG_IS_COMPRESSED) != 0) { // Check if MSB is
-																					// set
-			this.bodyData = decompressData();
-		} else {
-			this.bodyData = this.reader.readNextByteArray(this.scriptHeader.inflatedHeaderSize);
-		}
+		this.decompressionProvider = getDecompressionProvider();
+		this.blockHeader = new NsisBlockHeader(this.reader);
```

Comment:
The object `this.reader` will be the compressed bytes, not the decompressed bytes. I think you will need to do this:

```java
try (InputStream decompressedStream = this.getDecompressedInputStream()) {
  InputStreamByteProvider blockDataByteProvider = InputStreamByteProvider(decompressedStream);
  BinaryReader blockReader = new FactoryBundledWithBinaryReader(factory, blockDataByteProvider, NsisConstants.IS_LITTLE_ENDIAN);
  this.blockHeader = new NsisBlockHeader(blockReader);
  // Do more processing of the block data here
}
```

Reply:
Very good catch! I'll fix that